### PR TITLE
[Layout][Next.js] Load the content styles for the RichText component

### PR DESCRIPTION
**Pull request by illiakovalenko**
- [x] This PR follows the [Contribution Guide](https://github.com/Sitecore/jss/blob/dev/CONTRIBUTING.md)
- [x] Changelog updated

## Description / Motivation
* Introduced _getContentStylesheetLink_ function
* Next.js app fetches content styles in case **ck-content** class name is present in layout data
* When prefetching content styles on each page without traversing layout data, the page load time increases by approximately 50ms, whereas the layout data traversal only takes around 0.6ms
* We can't really identify if the field is RichText, so each field should be checked, if at least one field contains **ck-content** we can stop traversing and return a link
* In case layout data doesn't contain **ck-content** the content stylesheet link is null

## Testing Details

- [x] Unit Test Added
- [x] Manual Test/Other (Please elaborate)
Added a ck-content class under RichText field of my custom component

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/sitecore-jss/src/layout/content-styles.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  config: Config
+) => {
+  if (config.loadStyles) return;
+
+  components.forEach((component) => {
+    traverseComponent(component, config);
+  });
+};
+
+export const traverseField = (field: Field | Item | Item[] | undefined, config: Config) => {
```
Comment (sc-addypathania) [inline packages/sitecore-jss/src/layout/content-styles.ts:54]: should we define a separate type for these parameters?

File: packages/sitecore-jss/src/layout/content-styles.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  }
+};
+
+export const traverseComponent = (
+  component: RouteData | ComponentRendering | HtmlElementRendering,
+  config: Config
+) => {
+  if (config.loadStyles) return;
+
+  console.log(component);
```
Comment (sc-addypathania) [inline packages/sitecore-jss/src/layout/content-styles.ts:78]: remove the log statement

**Review by sc-addypathania — CHANGES_REQUESTED** [review]
Looks good.
See some comments.

**Review by sc-addypathania — CHANGES_REQUESTED** [review]
Looks good.
See some comments.

File: packages/sitecore-jss/src/layout/content-styles.ts (hunk 1 above)
Comment (illiakovalenko) [reply packages/sitecore-jss/src/layout/content-styles.ts:54]: Not really, it's just a list of different types of Fields. It's used across the project in a same way

**Review by sc-addypathania — APPROVED** [review]
👍🏼

**Pull request merged**